uwblueprint/starter-code-v2
# migrate entity service to prisma

## Notion ticket link

[Migrate Entity Service to Prisma](https://www.notion.so/uwblueprintexecs/Migrate-Entity-Service-to-Prisma-56cde56c97734f07b0f5741c9afedcde?pvs=25)



## Implementation description
* Refactor CRUD operations to help migrate the entity service from Sequelize to Prisma ORM. 


## Steps to test
1. From the CLI, run the command: "npm run dev -- --testing" and generate the PERN stack from the branch "rohan-sanskriti/migrate-entity-service"
2. Build the application by running the command: "docker-compose up --build -d"
3. From the Postman, under “Body” click raw and then “Json” and put your credentials:
{
        “email”: “user@example.com”,
        “password”: “your-password” 
}
6. Make a POST request to "http://localhost:8080/auth/login"
7. You will get the user token after the request which you then put under “Authorization” of type “Bearer Token”
8. Make the necessary GET, POST, and DELETE requests to "http://localhost:8080/entities" to create, update, and delete entities.

Note: if there are issues while testing from Postman, you can also go to localhost:3000 and create/update entities from there and access the PostgreSQL database to see whether the changes are reflected. You can do this by running the commands "docker exec -it scv2_db /bin/bash", then "psql -U postgres -d scv2", and then do "SELECT * FROM entities;" and you should see the entries.
